hera_qm/xrfi.py

```diff
@@ -1975,13 +1984,50 @@ def xrfi_run(ocalfits_file=None, acalfits_file=None, model_file=None, data_file=
                 'abscal_chi_sq_renormed_metrics2.h5': uvf_az2, 'abscal_chi_sq_flags2.h5': uvf_azf2,
                 'combined_metrics2.h5': uvf_metrics2, 'combined_flags2.h5': uvf_fws2,
                 'flags2.h5': uvf_combined2}
-    basename = qm_utils.strip_extension(os.path.basename(output_prefix))
-    for ext, uvf in uvf_dict.items():
-        if uvf is not None:
-            outfile = '.'.join([basename, ext])
-            outpath = os.path.join(dirname, outfile)
-            uvf.write(outpath, clobber=clobber)
 
+    # Determine the actual files to store
+    # We will drop kt_size / (integrations per file) files at the start and
+    # end to avoid edge effects from the convolution kernel.
+    # If this chunk includes the start or end of the night, we will write
+    # output files for those, but flag everything.
+
+    # Read metadata from first file to get integrations per file.
+    file_lists = [ocalfits_files, acalfits_files, data_files, model_files]
+    if data_files is not None:
+        dtypet = 'uvdata'
+        uvlist = data_files
+    elif model_files is not None:
+        dtypet = 'uvdata'
+        uvlist = model_files
+    elif ocalfits_files is not None:
+        dtypet = 'uvcal'
+        uvlist = ocalfits_files
+    elif acalfits_files is not None:
+        dtypet = 'uvcal'
+        uvlist = acalfits_files
+    if dtypet == 'uvdata':
+        uvtemp = UVData()
+        uvtemp.read(uvlist[0], read_data=False)
+    elif dtypet == 'uvcal':
+        uvtemp = UVCal()
+        uvtemp.read_calfits(uvlist[0])
+
+    # Loop through the files to output, storing all the different data products.
+    for ind in range(0, len(uvlist)):
+        dirname = resolve_xrfi_path(xrfi_path, output_prefixes[ind], jd_subdir=True)
+        basename = qm_utils.strip_extension(os.path.basename(output_prefixes[ind]))
+        for ext, uvf in uvf_dict.items():
+            if uvf is not None:
+                # This is calculated separately for each uvf because machine
+                # precision error was leading to times not found in object.
+                this_times = np.unique(uvf.time_array)
+                t_ind = ind * uvtemp.Ntimes
+                uvf_out = uvf.select(times=this_times[t_ind:(t_ind + uvtemp.Ntimes)],
```

Comment:
This assumes that all files have the same number of integrations. Do we think that might be a problem? Maybe this is a question for @plaplant?

I'd at least document the assumption and maybe raise an issue if we think it'll be a problem in the near term...

Reply:
I'll document it for now.